### ISSUE TITLE
Unable to use azure monitor for azurechina cloud

### ISSUE THREAD

**Issue by amitkumarcn** [grafana/grafana#22191]
**What happened**: Unable to query any metrics for azurechina using azure monitor plugin

**What you expected to happen**: should be able to query cpu/memory metrics for virtual machines

**How to reproduce it (as minimally and precisely as possible)**: try configure azure monitor datasource with azurechina cloud and plot any metrics for resources in azure china

**Anything else we need to know?**: 
Getting error:
t=2020-02-13T11:01:45+0000 lvl=info msg=Requesting logger=data-proxy-log url="https://management.chinacloudapi.cn/subscriptions/xxxxxxx/resourceGroups/yyyy/providers/Microsoft.Compute/virtualMachines/hostname/providers/microsoft.insights/metricdefinitions?api-version=2018-01-01&metricnamespace=Microsoft.Compute%2FvirtualMachines"
t=2020-02-13T11:01:48+0000 lvl=info msg="Got new access token" logger=data-proxy-log ExpiresOn=1970-01-01T00:00:00+0000
t=2020-02-13T11:01:48+0000 lvl=info msg=Requesting logger=data-proxy-log url=https://management.azure.com/chinaazuremonitor/subscriptions
t=2020-02-13T11:01:51+0000 lvl=eror msg="Request failed" logger=tsdb.azuremonitor status="404 Not Found" body="{\"error\":{\"code\":\"MissingSubscription\",\"message\":\"The request did not have a subscription or a valid tenant level resource provider.\"}}"
Why url: https://management.azure.com/chinaazuremonitor/subscriptions is being called.

**Environment**:
- Grafana version: 6.6.1
- Data source type & version: azure monitor
- OS Grafana is installed on: ubuntu 16.04
- User OS & Browser: chrome
- Grafana plugins: azure monitor
- Others:

**Comment by amitkumarcn** [comment]
Just to update here, i have tested this in Version 5.4.5 (commit: 091fd2c, branch: v5.4.x) and its working fine there.

**Comment by jumping** [comment]
the same issue

**Comment by uebenes** [comment]
I had the same issue in #23183 - its seems to be a problem only with the China Azure Monitoring API.
I have access to China Subscription of Azure, if you need help for testing.

**Comment by uebenes** [comment]
@daniellee 
It looks, that this issue is already known since 6 weeks. Is there any progress for fixing this issue.
The metrics API in China is different from the rest of world. This have to be considered in the code.

**Comment by uebenes** [comment]
sample:
![grafik](https://user-images.githubusercontent.com/59523839/77900452-a36f1c80-72b0-11ea-9387-0061528c8809.png)

**Comment by daniellee** [comment]
We don't have a way to test against the API for Azure China. I just merged #22753 but that fixes the issue for Azure Log Analytics and App Insights not for Azure Monitor.

I'm not seeing the url `https://management.azure.com/chinaazuremonitor/subscriptions` in my logs - that would be a bug.

Do the queries for the dropdowns work? For example, does the Resource Group dropdown show a list of resource groups?

**Comment by uebenes** [comment]
@daniellee 

Yes, the queried for the dropdown works perfecly and it automatically cound the subscriptions, the ressource groups and the devices. Thats why I was wondering, why the final API call wasn't successful.

**Comment by daniellee** [comment]
@uebenes in your Grafana server logs, can you see the url that is failing? You might have to change the logging level in your ini file to debug level for Azure Monitor to see it:

```
[log]
mode = console
filters = tsdb.azuremonitor:debug
level = info
```

**Comment by uebenes** [comment]
@daniellee 

I have added the config in to my server, but I see no logs in the console: 
Only: 2020-04-15T05:19:33.782104399Z Starting Grafana Server:.

I can fetch any logfiles, because its running in Docker. How can I access the grafana loginformation?

Thanks, 

Stefan

**Comment by amitkumarcn** [comment]
INFO[04-20|11:45:41] Requesting                               logger=data-proxy-log url="https://management.chinacloudapi.cn/subscriptions/xxxxxx/resourceGroups/rg/providers/Microsoft.Network/expressRouteCircuits/ERC/providers/microsoft.insights/metricdefinitions?api-version=2018-01-01&metricnamespace=Microsoft.Network%2FexpressRouteCircuits"
DBUG[04-20|11:45:42] AzureMonitor                             logger=tsdb.azuremonitor target="map[aggregation:Average allowedTimeGrainsMs:[60000 300000 900000 1800000 3600000 21600000 43200000 86400000] dimension:None dimensionFilter:* metricDefinition:Microsoft.Network/expressRouteCircuits metricName:BitsInPerSecond metricNamespace:Microsoft.Network/expressRouteCircuits resourceGroup:RG resourceName:ER timeGrain:auto top:10]"
DBUG[04-20|11:45:42] Scheduling update                        logger=alerting.scheduler ruleCount=0
INFO[04-20|11:45:43] Got new access token                     logger=data-proxy-log ExpiresOn=1970-01-01T05:30:00+0530
INFO[04-20|11:45:43] Requesting                               logger=data-proxy-log url=https://management.azure.com/chinaazuremonitor/subscriptions
DBUG[04-20|11:45:43] AzureMonitor                             logger=tsdb.azuremonitor Request ApiURL="https://management.azure.com/chinaazuremonitor/subscriptions/xxxx/resourceGroups/RG/providers/Microsoft.Network/expressRouteCircuits/ERC/providers/microsoft.insights/metrics?aggregation=Average&api-version=2018-01-01&interval=PT1M&metricnames=BitsInPerSecond&metricnamespace=Microsoft.Network%2FexpressRouteCircuits&timespan=2020-04-20T00%3A15%3A42Z%2F2020-04-20T06%3A15%3A42Z"
EROR[04-20|11:45:45] Request failed                           logger=tsdb.azuremonitor status="404 Not Found" body="{\"error\":{\"code\":\"MissingSubscription\",\"message\":\"The request did not have a subscription or a valid tenant level resource provider.\"}}"

Above logs were from grafana 6.6.2

**Comment by amitkumarcn** [comment]
I tried spoofing management.azure.com to china ip, call went fine, though azure rejected as destination certificate is applicable for management.chinacloudapi.cn hostname.

**Comment by uebenes** [comment]
@daniellee 

Is there any plan to fix this bug?

**Comment by zbbkeepgoing** [comment]
The same problem still exists in 6.7.3, Do we have a clear milestone about that.

**Comment by uebenes** [comment]
@daniellee : if your problem is only, that you can't test again a China Azure Cloud, I can tried to help. 
I have access to China Azure and could do the testing.

**Comment by uebenes** [comment]
Problem still existing in Grafana 7.0

![grafik](https://user-images.githubusercontent.com/59523839/83391006-64328880-a425-11ea-9e83-b944a31e3ba8.png)

**Comment by uebenes** [comment]
@daniellee : is there any solution ?

**Comment by JasonAlcoholic** [comment]
still no comment on this one?

**Comment by JasonAlcoholic** [comment]
still no comment on this one?

**Comment by daniellee** [comment]
So this is really hard to reproduce as we don't have access to an Azure China account. There is an attempted fix here for App Insights: https://github.com/grafana/grafana/pull/23877

Can someone provide debug logs to try and pinpoint which services (is it Azure Monitor, Log Analytics, App Insights or App Insights Analytics) and which API calls are not correct. That would be really useful.

**Comment by JasonAlcoholic** [comment]
Azure Monitor
![image](https://user-images.githubusercontent.com/5511205/85489296-60e48400-b602-11ea-97cf-fb03af749eea.png)
Some logs:
![image](https://user-images.githubusercontent.com/5511205/85489587-c9cbfc00-b602-11ea-9408-53384b00be07.png)

**Comment by jessie-pang** [comment]
@daniellee This issue has been pending for months, and it's really easy to reproduce in Azure China environment. Here is the information you are asking for:

1. Grafana version: 7.0.5
2. Service: Azure Monitor
3. Incorrect API call (Need to modify "**management.azure.com/chinaazuremonitor**" to "**management.chinacloudapi.cn**"): https://**management.azure.com/chinaazuremonitor**/subscriptions/xxxx/resourceGroups/RG/providers/Microsoft.Compute/virtualMachines/VMNAME/providers/microsoft.insights/metrics?aggregation=Average&api-version=2018-01-01&interval=PT1M&metricnames=Percentage+CPU&metricnamespace=Microsoft.Compute%2FvirtualMachines&timespan=2020-07-01T07%3A13%3A45Z%2F2020-07-01T13%3A13%3A45Z
4. Debug log is attached.
[Grafana_7_0_5_with_Azure_China.zip](https://github.com/grafana/grafana/files/4862437/Grafana_7_0_5_with_Azure_China.zip)

According to the debug log, the API calls to query subscriptions, resourceGroups, resources, metricNamespaces and metricdefinitions are all correct. They are pointing to management.chinacloudapi.cn.
Only the last API call to query metrics is using “management.azure.com/chinaazuremonitor” as the endpoint, which is incorrect.
 
https://management.chinacloudapi.cn/subscriptions?api-version=2019-03-01
https://management.chinacloudapi.cn/subscriptions/xxxx/resourceGroups?api-version=2018-01-01
https://management.chinacloudapi.cn/subscriptions/xxxx/resourceGroups/RG/resources?api-version=2018-01-01
https://management.chinacloudapi.cn/subscriptions/xxxx/resourceGroups/RG/providers/Microsoft.Compute/virtualMachines/VMNAME/providers/microsoft.insights/metricNamespaces?api-version=2017-12-01-preview
https://management.chinacloudapi.cn/subscriptions/xxxx/resourceGroups/RG/providers/Microsoft.Compute/virtualMachines/VMNAME/providers/microsoft.insights/metricdefinitions?api-version=2018-01-01&metricnamespace=Microsoft.Compute%2FvirtualMachines
https://management.azure.com/chinaazuremonitor/subscriptions
https://management.azure.com/chinaazuremonitor/subscriptions/xxxx/resourceGroups/RG/providers/Microsoft.Compute/virtualMachines/VMNAME/providers/microsoft.insights/metrics?aggregation=Average&api-version=2018-01-01&interval=PT1M&metricnames=Percentage+CPU&metricnamespace=Microsoft.Compute%2FvirtualMachines&timespan=2020-07-01T07%3A13%3A45Z%2F2020-07-01T13%3A13%3A45Z

**Comment by medmedchiheb** [comment]
Any updates on this particular issue!

**Comment by lolspider** [comment]
Any solutions?

**Comment by daniellee** [comment]
> it's really easy to reproduce in Azure China environment

Only if you have access to an Azure China environment.

I think this was fixed in Grafana 7.2: https://github.com/grafana/grafana/pull/27273 but it is impossible to test properly. Can anyone confirm if this is still an issue in Grafana 7.3.2?

@lolspider @medmedchiheb @jessie-pang

**Comment by lolspider** [comment]
@daniellee Great! I just installed Grafana 7.3.2 for testing, it works fine.

![image](https://user-images.githubusercontent.com/16553607/98931809-7b6bbb00-2519-11eb-8699-e5535af41f99.png)

**Comment by daniellee** [comment]
Thanks @lolspider 

Closing this issue as fixed by #27273

Can reopen if this is not solved for everyone.
